androidApp/src/main/java/com/example/tasky/android/agenda/viewmodel/AgendaDetailsViewModel.kt

```diff
@@ -110,6 +117,51 @@ class AgendaDetailsViewModel(
 
     private val deletedPhotoKeys = mutableListOf<String>()
 
+    init {
+        if (routeArguments.agendaId.isEmpty()) {
+            createLocalAgenda()
+        }
+    }
+
+    private fun createLocalAgenda() {
+        viewModelScope.launch {
+            val id = Uuid.random().toString()
+            val now = Clock.System.now().toEpochMilliseconds()
+            val remindAt = now - 10.minutes.toLong(DurationUnit.MILLISECONDS)
+
+            _screenStateFlow.update {
+                it.copy(
+                    agendaItem =
+                        when (routeArguments.type) {
+                            AgendaDetailsScreenType.TASK -> Task.EMPTY.copy(id = id, time = now, remindAt = remindAt)
```

Comment:
I found it better to use an enum through my presentation and domain layer to represent when the user wants an alarm to go off for an agenda item. Then, in my data layer, I calculate the Long value to send to the API.

With this approach, I don't perform any calculations (to get a new Long value) when the user changes the reminder time. I also get the advantage of working with a higher-level object like an enum instead of a Long.